enhancements/network/on-prem-configurable-lb-dns-stack.md

```diff
@@ -0,0 +1,284 @@
+
+#### Disaster recovery support
+
+In order to support disaster recovery kubelet, kube-controller-manager and kube-scheduler should be able to communicate with kube-api even when all cluster-hosted-net-services-operator pods are shutdown.
+
+The following steps will ensure that kube-api is available post disaster recovery:
+
+- Dnsmasq service and Keepalived static pods will start running on master nodes
+- API-VIP address will be assigned to one of the master nodes by Keepalived
+- Dnsmasq should resolve api-int hostname to API-VIP address
+
+The above steps should be applied also for cluster shutdown and resume use-case.
+
+### cluster-hosted-net-services operator
+
+The `cluster-hosted-net-services-operator` is a new operator whose operand is a controller for the cluster hosted DNS, LB and VIP services.
+
+### Standard SLO Behavior
+
+The cluster-hosted-net-services-operator is expected to follow the standard behavior of SLO, additionally, since the cluster-hosted-net-services-operator is applicable only to clusters running on on-prem ( 'baremetal', 'openstack', 'vsphere' or 'ovirt') infrastructure,
+It should support also the expected behaviors for "not in use" SLOs similar to the [CBO operator](https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/an-slo-for-baremetal.md#standard-slo-behaviors).
+
+### cluster-hosted-net-services-operator details
+
+The cluster-hosted-net-services-operator should:
+
+- Be a new `openshift/cluster-hosted-net-services-operator` project.
+- Publish an image called `cluster-hosted-net-services-operator`.
+- Add a new `ClusterOperator` with an additional
+  `Disabled` status for non on-prem platforms.
+- Implement a controller reconciling a singleton instance representing the desired cluster hosted LB/DNS/VIP configuration.
+- Do nothing except set `Disabled=true`, `Available=true`, and
+  `Progressing=False` when the `Infrastructure` resource is a platform
+  type other than openstack, baremetal, ovirt or vsphere.
+- Update `ClusterOperator` DEGRADED field in accordance with the following healthchecks (in case self-hosted stack is enabled):
+  - single node holds the VIP
+  - kube-api is available via the VIP
+
+#### CRD definition
+
+Below is a possible option for a CRD instance of this operator:
+
+```yaml
+apiVersion: clusterHostedNetServices.operator.io/v1alpha1
+kind: HostedNetServices
+metadata:
+  name: cluster
+spec:
+  dns: 
+    nodesResolution: Enabled
+    appsResolution: Enabled
+    apiintIpAddress: 192.168.111.5 
+  ingressHa: Enabled
+status:
+  apiServerInternalIpOwner: master-0
+  ingressIpOwner: worker-1
+```
+
+With this sample manifest the cluster-hosted-net-services-operator should:
+
+- Create DNS hosted service that will:
+
+  - resolve node names and .apps wildcard record
+  - resolve api-int to 192.168.111.5.
+  - Run the self-hosted Loadbalancer for api only if apiintIpAddress is equal to API-VIP value provides in install-config file.
```

Comment:
what is that API-VIP value during cluster runtime?

Reply:
The VIP address provided in the install-config.yaml file